Plugins/GitUIPluginInterfaces/ApplicationSettings.cs

```diff
@@ -0,0 +1,52 @@
+﻿namespace GitUIPluginInterfaces
```

Comment:
This adds several layers of abstraction, and I personally can't see any benefits for doing this. I would appreciate it if you could detail what the pros and cons of your design proposal, and why it is better than, say, something like this:
```cs
    public class DetailedSettings
    {
        public static readonly string SmtpServerDefault = "smtp.gmail.com";
        public static readonly int SmtpPortDefault = 465;
        public static readonly bool SmtpUseSslDefault = true;

        public DetailedSettings(ISettingsSource settingsSource)
        {
            ...
        }

        public string SmtpServer { get => ...; set => ...; } = SmtpServerDefault;
        public int SmtpPort { get => ...; set => ...; } = SmtpPortDefault;
        public bool SmtpUseSsl { get => ...; set => ...; } = SmtpUseSslDefault;
    }
```

Reply:
I thought so. 
I left the constants separately, if they are needed for access, since the implementation will be internal.

`internal sealed class DetailedSettings : IDetailedSettings`

Reply:
Moved to the DetailedSettings class.